ThreeBotPackages/threefold/directory/actors/farms.py

```diff
@@ -6,39 +6,53 @@ def _init(self, **kwargs):
         bcdb = j.data.bcdb.get("tf_directory")
         self.farm_model = bcdb.model_get(url="tfgrid.farm.1")
 
-    def register(self, farm):
+    def _get_farm(self, farm_id):
+        try:
+            return self.farm_model.get(farm_id)
+        except j.exceptions.NotFound:
+            raise j.exceptions.NotFound("farm %s not found" % farm_id)
+
+    def register(self, farm, schema_out):
         """
         ```in
         farm = (O) !tfgrid.farm.1
         ```
+
+        ```out
+        farm_id = (I)
```

Comment:
I'm not yet sure how we're going to handle the identity of the farm. This issue discussed that: https://github.com/threefoldtech/tf_cockpit/issues/12

We should adapt the logic according to what is decided in the issue